t/t5555-http-smart-common.sh

```diff
@@ -123,9 +123,19 @@ test_expect_success 'git receive-pack --advertise-refs: v1' '
 '
 
 test_expect_success 'git upload-pack --advertise-refs: v2' '
+	printf "agent=FAKE" >agent_os_name &&
+	if test_have_prereq WINDOWS
+	then
+		test_config transfer.advertiseOSVersion false
```

Comment:
After thinking again about it, I changed my mind and my opinion now is that it's probably better to always use `git config` instead of `test_config` when disabling transfer.advertiseOSVersion on Windows. Otherwise the following tests cannot easily reuse the files that are created with `os-version=...` as we saw with t5701 below. They would all have to disable transfer.advertiseOSVersion on Windows first which would be tedious if there are multiple tests that want to reuse the files.

It might be worth telling in a comment that we use `git config` instead of `test_config` so that other tests below can reuse the 'expect' file though.

Reply:
Yeah, this is well thought out. It is very true. Using test_config will lead to repetition. I will change it to git config.